### PR TITLE
[api] Add /guess_format and /guess_field_types public operation

### PR DESCRIPTION
**Pull request by agl29**
## What changes were proposed in this pull request?

- Adding /guess_format and /guess_field_types apis

### CODE REVIEW

File: desktop/core/src/desktop/api_public_urls.py [hunk 1]
```diff
@@ -88,3 +88,8 @@
 urlpatterns += [
   re_path(r'^slack/install/?$', botserver_api.generate_slack_install_link, name='botserver.api.slack_install_link'),
 ]
+
+urlpatterns += [
+  re_path(r'^indexer/guess_format/?$', api_public.guess_format, name='guess_format'),
+  re_path(r'^indexer/guess_field_types/?$', api_public.guess_field_types, name='guess_field_types'),
```
Comment (romainr) [inline desktop/core/src/desktop/api_public_urls.py:94]: Prefix name with `indexer_`?

Comment (agl29) [reply desktop/core/src/desktop/api_public_urls.py]: done

**Review by romainr — APPROVED** [review]
Update the API docs too now?

**Comment by agl29** [comment]
Now, tests are passed for py3 but not for py2. So still import error (ss)?
Screenshot 2021-07-07 at 6 42 49 PM

**Comment by agl29** [comment]
Now no circular dependency -> https://stackoverflow.com/questions/9252543/importerror-cannot-import-name-x
and https://stackoverflow.com/a/746067/13946851

is any other method to import in this condition? @romainr

**Comment by agl29** [comment]
test results for apis -> 
Screenshot 2021-07-07 at 9 43 40 PM
Screenshot 2021-07-07 at 9 44 56 PM

File: desktop/libs/kafka/src/kafka/kafka_api.py [hunk 2]
```diff
@@ -20,7 +20,7 @@
 import logging
 import sys
 
-from desktop.api_public import _get_interpreter_from_dialect
+from desktop.api_public import *
```
Comment (romainr) [inline desktop/libs/kafka/src/kafka/kafka_api.py:23]: Please avoid * and put the explicit names

**Review by romainr — COMMENTED** [review]
Nice! Just one comment

File: desktop/libs/kafka/src/kafka/kafka_api.py (hunk 2 above)
Comment (agl29) [reply desktop/libs/kafka/src/kafka/kafka_api.py:23]: but here if i put the explicit name then I am getting this error `ImportError: cannot import name '_get_interpreter_from_dialect' from partially initialized module 'desktop.api_public' (most likely due to a circular import)`

**Comment by romainr** [comment]
No import *, if circular just inline it like in other places

File: desktop/libs/indexer/src/indexer/indexers/flink_sql.py [hunk 3]
```diff
@@ -47,6 +46,7 @@ def create_table_from_kafka(self, source, destination, start_time=-1, dry_run=Fa
 
     source_type = source['sourceType']
 
+    from desktop.api_public import _get_interpreter_from_dialect   # due to a circular import
```
Comment (agl29) [inline desktop/libs/indexer/src/indexer/indexers/flink_sql.py:49]: @romainr  is it good now?

Comment (romainr) [reply desktop/libs/indexer/src/indexer/indexers/flink_sql.py:49]: (and we will see later if we can o better)

**Review by romainr — APPROVED** [review]

**Pull request merged**